commit 33e26f3544a558e7476eb221ff33173759b3a116
Merge: 16893dd23f6d b7b68c6e3677
Author: Dave Airlie <airlied@redhat.com>
Date:   Thu Feb 27 10:08:22 2025 +1000

    Merge tag 'drm-xe-next-2025-02-24' of https://gitlab.freedesktop.org/drm/xe/kernel into drm-next
    
    UAPI Changes:
     - Add mmap support for PCI memory barrier (Tejas, Matthew Auld)
     - Enable integration with perf pmu, exposing event counters: for now, just
       GT C6 residency (Vinay, Lucas)
     - Add "survivability mode" to allow putting the driver in a state capable of
       firmware upgrade on critical failures (Riana, Rodrigo)
     - Add PXP HWDRM support and enable for compatible platforms:
       Meteor Lake and Lunar Lake (Daniele, John Harrison)
     - Expose package and vram temperature over hwmon subsystem (Raag, Badal, Rodrigo)
    
    Cross-subsystem Changes:
     - Backmege drm-next to synchronize with i915 display and other internal APIs
    
    Display Changes (including i915):
     - Device probe re-order to help with flicker-free boot (Maarten)
     - Align watermark, hpd and dsm with i915 (Rodrigo)
     - Better abstraction for d3cold (Rodrigo)
    
    Driver Changes:
     - Make sure changes to ccs_mode is with helper for gt sync reset (Maciej)
     - Drop mmio_ext abstraction since it didn't prove useful in its current form
       (Matt Roper)
     - Reject BO eviction if BO is bound to current VM (Oak, Thomas Hellström)
     - Add GuC Power Conservation debugfs (Rodrigo)
     - L3 cache topology updates for Xe3 (Francois, Matt Atwood)
     - Better logging about missing GuC logs (John Harrison)
     - Better logging for hwconfig-related data availability (John Harrison)
     - Tracepoint updates for xe_bo_create, xe_vm and xe_vma (Oak)
     - Add missing SPDX licenses (Francois)
     - Xe suballocator imporovements (Michal Wajdeczko)
     - Improve logging for native vs SR-IOV driver mode (Satyanarayana)
     - Make sure VF bootstrap is not attempted in execlist mode (Maarten)
     - Add GuC Buffer Cache abstraction for some CTB H2G actions and use
       during VF provisioning (Michal Wajdeczko)
     - Better synchronization in gtidle for new users (Vinay)
     - New workarounds for Panther Lake (Nirmoy, Vinay)
     - PCI ID updates for Panther Lake (Matt Atwood)
     - Enable SR-IOV for Panther Lake (Michal Wajdeczko)
     - Update MAINTAINERS to stop directing xe changes to drm-misc (Lucas)
     - New PCI IDs for Battle Mage (Shekhar)
     - Better pagefault logging (Francois)
     - SR-IOV fixes and refactors for past and new platforms (Michal Wajdeczko)
     - Platform descriptor refactors and updates (Sai Teja)
     - Add gt stats debugfs (Francois)
     - Add guc_log debugfs to dump to dmesg (Lucas)
     - Abstract per-platform LMTT availability (Piotr Piórkowski)
     - Refactor VRAM manager location (Piotr Piórkowski)
     - Add missing xe_pm_runtime_put when forcing wedged mode (Shuicheng)
     - Fix possible lockup when forcing wedged mode (Xin Wang)
     - Probe refactors to use cleanup actions with better error handling (Lucas)
     - XE_IOCTL_DBG clarification for userspace (Maarten)
     - Better xe_mmio initialization and abstraction (Ilia)
     - Drop unnecessary GT lookup (Matt Roper)
     - Skip client engine usage from fdinfo for VFs (Marcin Bernatowicz)
     - Allow to test xe_sync_entry_parse with error injection (Priyanka)
     - OA fix for polled read (Umesh)
    
    Signed-off-by: Dave Airlie <airlied@redhat.com>
    
    From: Lucas De Marchi <lucas.demarchi@intel.com>
    Link: https://patchwork.freedesktop.org/patch/msgid/m3gbuh32wgiep43i4zxbyhxqbenvtgvtao5sczivlasj7tikwv@dmlba4bfg2ny

diff --cc drivers/gpu/drm/xe/display/xe_display.c
index 96ba9595bf2a,7fef78f5606d..02a413a07382
--- a/drivers/gpu/drm/xe/display/xe_display.c
+++ b/drivers/gpu/drm/xe/display/xe_display.c
@@@ -175,13 -167,12 +167,12 @@@ static void xe_display_fini(void *arg
  	struct xe_device *xe = arg;
  	struct intel_display *display = &xe->display;
  
- 	if (!xe->info.probe_display)
- 		return;
- 
- 	intel_display_driver_remove_nogem(display);
+ 	intel_hpd_poll_fini(xe);
+ 	intel_hdcp_component_fini(display);
 -	intel_audio_deinit(xe);
++	intel_audio_deinit(display);
  }
  
- int xe_display_init_noaccel(struct xe_device *xe)
+ int xe_display_init(struct xe_device *xe)
  {
  	struct intel_display *display = &xe->display;
  	int err;